numpy/distutils/ccompiler_opt.py

```diff
@@ -268,6 +268,7 @@ class _Config:
             extra_checks="AVX512F_REDUCE"
         ),
         AVX512CD = dict(interest=21, implies="AVX512F"),
+        AVX512FP16 = dict(interest=22, implies="AVX512F"),
```

Comment:
```suggestion
```
no need for adding a separate target for `AVX512FP16`, having `AVX512_SPR` is just enough since it's only limited to this CPU family.